src/librawspeed/common/Common.h

```diff
@@ -177,10 +180,9 @@ extern "C" { FILE __iob_func[3] = { *stdin,*stdout,*stderr }; }
 #endif
 
 // clampBits clamps the given int to the range 0 .. 2^n-1
-inline uint32 clampBits(int x, uint32 n)
-{
-  const int tmp = (1 << n) - 1;
-  x = x < 0 ? 0 : x;
+inline uint32 clampBits(int64 x, uint32 n) {
```

Comment:
It seems CID 1402217 is finding it problematic that you upgraded x from 'int' to 'int64'. I was about to question that before, since the code has no use for that (it gets called with an int and n is always less than 17). It seemed to me you wanted to support n up to 32 and thought "why complain, it should not hurt". Just saying: the easy fix for the CID seems to go back to `int` and limit your tests to a max of `n == 31`. Maybe add an `assert(n < 32);` for good measure.